test/WebJobs.Script.Tests/FileProvisioning/PowerShellFileProvisionerTests.cs

```diff
@@ -2,41 +2,135 @@
+using Microsoft.Extensions.Logging;
 using Xunit;
 
 namespace Microsoft.Azure.WebJobs.Script.Tests.FileAugmentation
 {
     public class PowerShellFileProvisionerTests
     {
+        private const string ProfilePs1FileName = "profile.ps1";
+        private const string RequirementsPsd1FileName = "requirements.psd1";
+
         private readonly string _scriptRootPath;
+        private TestLogger _logger;
 
         public PowerShellFileProvisionerTests()
         {
+            _logger = new TestLogger("PowerShellFileProvisionerTests");
             _scriptRootPath = Path.GetTempPath();
         }
 
         [Fact]
         public async Task AugmentFiles_Test()
         {
-            File.Delete(Path.Combine(_scriptRootPath, "requirements.psd1"));
-            File.Delete(Path.Combine(_scriptRootPath, "profile.ps1"));
-            var powerShellFileProvisioner = new PowerShellFileProvisioner();
+            EnsurePowerShellFilesDoNotExist(_scriptRootPath);
+
+            var powerShellFileProvisioner = new PowerShellFileProvisionerMocksPSGalleryCalls(_logger);
+            await powerShellFileProvisioner.ProvisionFiles(_scriptRootPath);
+
+            File.Exists(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+            File.Exists(Path.Combine(_scriptRootPath, ProfilePs1FileName));
+
+            string requirementsContent = File.ReadAllText(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+
+            string expectedContent = @"# This file enables modules to be automatically managed by the Functions service.
+# Only the Azure Az module is supported in public preview.
+# See https://aka.ms/functionsmanageddependency for additional information.
+#
+@{
+    # For latest supported version, go to 'https://www.powershellgallery.com/packages/Az'. 
+    'Az' = '2.*'
+}";
+            Assert.Equal(requirementsContent, expectedContent, StringComparer.OrdinalIgnoreCase);
+
+            ValidateLogs(_logger);
+        }
+
+        [Fact]
+        public async Task AugmentFiles_UnableToReachThePSGallery_Test()
+        {
+            EnsurePowerShellFilesDoNotExist(_scriptRootPath);
+
+            var powerShellFileProvisioner = new PowerShellFileProvisionerMocksPSGalleryCalls(_logger);
+            powerShellFileProvisioner.GetLatestAzModuleMajorVersionThrowsException = true;
             await powerShellFileProvisioner.ProvisionFiles(_scriptRootPath);
-            File.Exists(Path.Combine(_scriptRootPath, "requirements.psd1"));
-            File.Exists(Path.Combine(_scriptRootPath, "profile.ps1"));
+
+            File.Exists(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+            File.Exists(Path.Combine(_scriptRootPath, ProfilePs1FileName));
+
+            string requirementsContent = File.ReadAllText(Path.Combine(_scriptRootPath, RequirementsPsd1FileName));
+
+            string expectedContent = @"# This file enables modules to be automatically managed by the Functions service.
```

Comment:
Same here:  eigher read contents of src/WebJobs.Script/FileProvisioning/PowerShell/requirements.psd1 or move this string to constants

Reply:
I will move the string to constants.